tracer/tracer.go

```diff
@@ -1180,6 +1295,34 @@ func (t *Tracer) StartProbabilisticProfiling(ctx context.Context) {
 	})
 }
 
+// StartOffCPUProfiling starts off-cpu profiling by attaching the programs to the hooks.
+func (t *Tracer) StartOffCPUProfiling() error {
+	// Attach the second hook for off-cpu profiling first.
+	kprobeProg, ok := t.ebpfProgs["finish_task_switch"]
+	if !ok {
+		return errors.New("off-cpu program finish_task_switch is not available")
+	}
+
+	kprobeLink, err := link.Kprobe("finish_task_switch.isra.0", kprobeProg, nil)
```

Comment:
GCC 10 [introduced](https://gcc.gnu.org/gcc-10/changes.html) a breaking change with IPA-SRA. So depending on the compiler version, that was used to compile the Linux kernel, the symbol is either `finish_task_switch` or `finish_task_switch.isra.X`. For the concept implementation `finish_task_switch.isra.0` is used. To improve and provide backwards compatibility there are the following options:
- lookup kernel symbols before attaching and return the symbol, that starts with `finish_task_switch`
- use a hard coded list of symbols and try them one by one - e.g. `finish_task_switch.isra.0`, `finish_task_switch.isra.1` and `finish_task_switch`

What is the preferred way forward here?

Reply:
Suggestion:
Try for `finish_task_switch` first. If not found, try for a symbol starting with `finish_task_switch.isra.`

Reply:
> Try for finish_task_switch first [..]

My recommendation would be to lookup for a symbol with the prefix `finish_task_switch.isra.` first, before falling back to `finish_task_switch`. `finish_task_switch` exists just for backwards compatibility and even Debian already moved on and is using `finish_task_switch.isra.0`. As the `0` is not fixed and could be any other positive number, hardcoding a list of possible symbols, should also not preferred.

Reply:
How about searching the kernel symbols list (`/proc/kallsyms`)  which we already load and parse? We can easily find symbols with pattern matching there.

Reply:
Add the dynamic approach with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/196/commits/f27fa68807007b0f6c0c5bab67d218588e1ec71f